### PR TITLE
adding code_with_match (#1024)

### PR DESCRIPTION
**Pull request by QuentinRa**
Hi! As described in #1024, I added a function `code_with_match`.

### CODE REVIEW

File: lib/Github/Api/Search.php [hunk 1]
```diff
@@ -61,6 +61,25 @@ public function code($q, $sort = 'updated', $order = 'desc')
...
+     *
+     * @link https://docs.github.com/en/rest/reference/search#text-match-metadata
+     *
+     * @param string $q     the filter
+     * @param string $sort  the sort field
+     * @param string $order asc/desc
+     *
+     * @return array list of code found
+     */
+    public function code_with_match($q, $sort = 'updated', $order = 'desc')
```
Comment (acrobat) [inline lib/Github/Api/Search.php:76]:
```suggestion
    public function codeWithMatch(string $q, string $sort = 'updated', string $order = 'desc')
```

Also remove the redundant phpdoc after this change
```
     * @param string $q     the filter
     * @param string $sort  the sort field
     * @param string $order asc/desc
```

File: doc/search.md [hunk 2]
```diff
@@ -13,13 +13,19 @@ $repos = $client->api('search')->repositories('github language:php');
...
- 
+
 ```php
 $files = $client->api('search')->code('@todo language:php');
 ```
 
 Returns a list of files found by such criteria (containing "@todo" and language==php).
 
+```php
+$files = $client->api('search')->code_with_match('@todo language:php');
```
Comment (acrobat) [inline doc/search.md:24]: Update the method after the change below

**Review by acrobat — CHANGES_REQUESTED** [review]
Thanks for the PR @QuentinRa! Few minor comments and can you add a test for this method? See the related code search for a test example

https://github.com/KnpLabs/php-github-api/blob/de92500d4ca47743d81f39296f74104891ca88de/test/Github/Tests/Api/SearchTest.php#L93-L111

**Comment by akehsanz** [comment]
UNSUBSCRIBING to this thread. but all emails arrive like a Deluge...

**Comment by QuentinRa** [comment]
Thanks! I looked for tests, and for CONTRIBUTING.md/code_style.*, but I haven't found any (I should've checked by myself, instead of relying on my IDE 😖).

I added the return type too. I'm not sure if I should remove the "redundant" return type in the DOC, my IDE is showing a warning 😶"Update PHPDoc comment - Argument type do not match the declared type" (PHPStorm).

I'm not sure if the tests are what you wanted, I can't run them at all (and I don't have much time to look into it).

File: lib/Github/Api/Search.php [hunk 3]
```diff
@@ -61,6 +61,25 @@ public function code($q, $sort = 'updated', $order = 'desc')
...
 
+    /**
+     * Search code by filter (q), but will return additional data to highlight
+     * the matched results.
+     *
+     * @link https://docs.github.com/en/rest/reference/search#text-match-metadata
+     *
+     * @param $q     the filter
+     * @param $sort  the sort field
+     * @param $order asc/desc
```
Comment (acrobat) [inline lib/Github/Api/Search.php:72]: Please remove these phpdocs as the are redundant with the added typehint's

**Review by acrobat — CHANGES_REQUESTED** [review]
I comment left based on the failing phpstan run. Otherwise the PR is good to go!

**Review by acrobat — APPROVED** [review]

**Comment by acrobat** [comment]
Thanks @QuentinRa! And congrats on your first contribution! 🎉

**Pull request merged**